src/documentation/markdown.rs

```diff
@@ -382,4 +418,107 @@ And a line, or a rule."##;
             });
         assert_eq!(v.first(), Some(&(12..14)));
     }
+
+    #[test]
+    fn markdown_reduction_mapping_inline_link() {
+        const MARKDOWN: &str = r#" dyrck [I'm an inline-style link](https://www.google.com) artic"#;
+        const PLAIN: &str = r#"dyrck I'm an inline-style link artic"#;
+
+        let (plain, mapping) = PlainOverlay::extract_plain_with_mapping(MARKDOWN);
+
+        assert_eq!(dbg!(&plain).as_str(), PLAIN);
+        assert_eq!(dbg!(&mapping).len(), 3);
+        for (reduced_range, markdown_range) in mapping.iter() {
+            println!("{:?} {:?}", reduced_range, markdown_range);
+            assert_eq!(
+                plain[reduced_range.clone()].to_owned(),
+                MARKDOWN[markdown_range.clone()].to_owned()
+            );
+        }
+    }
+
+    #[test]
+    fn markdown_reduction_mapping_auto_link() {
+        const MARKDOWN: &str = r#" <http://foo.bar/baz>"#;
+        const PLAIN: &str = r#""#;
+
+        let (plain, mapping) = PlainOverlay::extract_plain_with_mapping(MARKDOWN);
+
+        assert_eq!(dbg!(&plain).as_str(), PLAIN);
+        assert_eq!(dbg!(&mapping).len(), 0);
+        for (reduced_range, markdown_range) in mapping.iter() {
+            assert_eq!(plain[reduced_range.clone()].to_owned(), String::from(""));
+        }
+    }
+
+    #[test]
+    fn markdown_reduction_mapping_auto_link_with_misspelled_words() {
+        const MARKDOWN: &str = r#" dyrck <http://foo.bar/baz> artic"#;
+        const PLAIN: &str = r#"dyrck  artic"#;
+
+        let (plain, mapping) = PlainOverlay::extract_plain_with_mapping(MARKDOWN);
+
+        assert_eq!(dbg!(&plain).as_str(), PLAIN);
+        assert_eq!(dbg!(&mapping).len(), 2);
+        for (reduced_range, markdown_range) in mapping.iter() {
+            assert_eq!(
+                plain[reduced_range.clone()].to_owned(),
+                MARKDOWN[markdown_range.clone()].to_owned()
+            );
+        }
+    }
+
+    #[test]
+    fn markdown_reduction_mapping_email_with_misspelled_words() {
+        const MARKDOWN: &str = r#" dyrck <joe@example.com> artic"#;
+        const PLAIN: &str = r#"dyrck  artic"#;
+
+        let (plain, mapping) = PlainOverlay::extract_plain_with_mapping(MARKDOWN);
+
+        assert_eq!(dbg!(&plain).as_str(), PLAIN);
+        assert_eq!(dbg!(&mapping).len(), 2);
+        for (reduced_range, markdown_range) in mapping.iter() {
+            assert_eq!(
+                plain[reduced_range.clone()].to_owned(),
+                MARKDOWN[markdown_range.clone()].to_owned()
+            );
+        }
+    }
+
+    #[test]
+    #[ignore]
+    fn markdown_reduction_mapping_footnote() {
+        // What is a markdown footnote syntax? Example:
```

Comment:
@drahnr I just wrote here my thoughts on this also in the pr description. Please, share your thoughts. Thanks!